### PR TITLE
Update/request unit test

### PR DESCRIPTION
**Pull request by JWZeta**

### CODE REVIEW

File: include/common_table.h [hunk 1]
```diff
@@ -38,7 +38,7 @@ typedef struct CommonTable
 } Table;
 
 typedef enum
-{ KEY, SERVER } TableType;
+{ KEY, SERVER, TEST } TableType;
```
Comment (b-carr) [inline include/common_table.h:41]: What's the purpose of the ```TEST``` type? I think the only use of it I noticed was in a test of ```table_destroy```.

Comment (JWZeta) [reply include/common_table.h:41]: This is a test type to be used in the test of ```table_destroy``` and later in other table unit test if needed.

File: test/src/util/table_test_suite.c [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+
+  //Initial load of certs into the server table
+  CU_ASSERT(read_bytes_from_file((char *) valid_id[6], &data, &data_len) == 0);
+  CU_ASSERT(kmyth_sgx_unseal_nkl(eid, data, data_len, &handle) == 0);
+  free(data);
+  server_table_add(eid, &status, handle);
+  CU_ASSERT(status == OK);
+
+  CU_ASSERT(read_bytes_from_file((char *) valid_id[7], &data, &data_len) == 0);
+  CU_ASSERT(kmyth_sgx_unseal_nkl(eid, data, data_len, &handle) == 0);
```
[the remark this answers is not in the document]
Comment (JWZeta) [reply test/src/util/table_test_suite.c]: Changed to check

File: test/src/util/table_test_suite.c [hunk 3]
```diff
@@ -0,0 +1,365 @@
...
+    memcpy(key.chars, key_str[i], key.len);
+    key_table_add_key(eid, &status, tmp, key);
+    CU_ASSERT(status == OK);
+    free_charbuf(&tmp);
+    secure_free_charbuf(&key);
+  }
+
+  //Initial load of certs into the server table
+  CU_ASSERT(read_bytes_from_file((char *) valid_id[6], &data, &data_len) == 0);
+  CU_ASSERT(kmyth_sgx_unseal_nkl(eid, data, data_len, &handle) == 0);
```
[the remark this answers is not in the document]
Comment (JWZeta) [reply test/src/util/table_test_suite.c]: Changed to check

File: test/src/util/table_test_suite.c [hunk 4]
```diff
@@ -0,0 +1,365 @@
...
```
[the remark this answers is not in the document]
Comment (JWZeta) [reply test/src/util/table_test_suite.c]: Changed to check

File: test/src/util/table_test_suite.c [hunk 5]
```diff
@@ -0,0 +1,365 @@
...
+  free_charbuf(&tmp);
+  secure_free_charbuf(&key);
+  pelz_log(LOG_INFO, "Key Table add Key success");
+
+  tmp = copy_CWD_to_id(prefix, valid_id[1]);
+  key_table_add_from_handle(eid, &status, tmp, handle);
+  CU_ASSERT(status == RET_FAIL);
+
+  CU_ASSERT(read_bytes_from_file((char *) valid_id[1], &data, &data_len) == 0);
+  CU_ASSERT(kmyth_sgx_unseal_nkl(eid, data, data_len, &handle) == 0);
```
[the remark this answers is not in the document]
Comment (JWZeta) [reply test/src/util/table_test_suite.c]: Changed to check

File: test/src/util/table_test_suite.c [hunk 6]
```diff
@@ -0,0 +1,365 @@
...
+  key = new_charbuf(strlen(key_str));
+  memcpy(key.chars, key_str, key.len);
+  key_table_add_key(eid, &status, tmp, key);
+  CU_ASSERT(status == OK);
+  free_charbuf(&tmp);
+  secure_free_charbuf(&key);
+  pelz_log(LOG_INFO, "Key Table add Key success");
+
+  tmp = copy_CWD_to_id(prefix, valid_id[1]);
+  key_table_add_from_handle(eid, &status, tmp, handle);
```
[the remark this answers is not in the document]
Comment (JWZeta) [reply test/src/util/table_test_suite.c:93]: Initialized to 0 at declaration

File: test/src/util/request_test_suite.c [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+  const char *tmp_id[2] = { "/test/data/key7.txt", "/test/data/key1txt" };
+  const char *key_str[3] = { "KIENJCDNHVIJERLMALIDFEKIUFDALJFG", "KALIENGVBIZSAIXKDNRUEHFMDDUHVKAN", "HVIJERLMALIDFKDN" };
+
+  //Initial data_in values
+  data_in = new_charbuf(32);
+  memcpy(data_in.chars, "abcdefghijklmnopqrstuvwxyz012345", data_in.len);
+
+  pelz_log(LOG_DEBUG, "Test Request Function Start");
+
+  //KEK not loaded so function should return KEK_LOAD_ERROR
```
[the remark this answers is not in the document]
Comment (JWZeta) [reply test/src/util/request_test_suite.c]: Changed to comment to KEK_NOT_LOADED

File: src/util/key_table.c [hunk 8]
```diff
@@ -117,7 +117,7 @@ TableResponseStatus key_table_add_from_server(charbuf key_id, charbuf server_id,
   if (ret)
   {
     pelz_log(LOG_ERR, "Retrieve Key function failure");
-    return ERR;
+    //return ERR;
```
[the remark this answers is not in the document]
Comment (b-carr) [reply src/util/key_table.c:120]: Got it, so at the moment ```key_table_add_key_from_server``` just adds a dummy key, but once ```enclave_retrieve_key``` is fully implemented it'll add the real key.

**Review by b-carr — APPROVED** [review]

**Pull request merged**